packages/firestore/src/remote/watch_change.ts

```diff
@@ -507,23 +511,40 @@ export class WatchChangeAggregator {
       } else {
         logWarn('Applying bloom filter failed: ', err);
       }
-      return BloomFilterApplicationStatus.Skipped;
+      return { status: BloomFilterApplicationStatus.Skipped };
     }
 
+    const bloomFilterMightContain = (documentPath: string): boolean => {
+      const databaseId = this.metadataProvider.getDatabaseId();
+      return bloomFilter.mightContain(
+        `projects/${databaseId.projectId}/databases/${databaseId.database}` +
+          `/documents/${documentPath}`
+      );
+    };
+
     if (bloomFilter.bitCount === 0) {
```

Comment:
why we want to pass the `bloomFilterMightContain` around as a parameter?   If we want to differentiate the situations which we should call `createExistenceFilterMismatchInfoForTestingHooks`, can we  do it based on the apply status instead? Like if the status is not skipped, then pass it into `createExistenceFilterMismatchInfoForTestingHooks`:
```
createExistenceFilterMismatchInfoForTestingHooks(
              applyResult.status,
              applyStatus !== BloomFilterApplicationStatus.Skipped ? applyResult.bloomFilterMightContain : null,
              currentSize,
              watchChange.existenceFilter
            )
```

Then the only exception will be `if (bloomFilter.bitCount === 0) ` which also returns Skipped. should this be treated as mismatch?

Reply:
Are you saying that we shouldn't pass around `bloomFilterMightContain` if `bloomFilter.bitCount === 0`? If so, I actually totally agree because the bloom filter is effectively "useless" when bitCount===0. I'll make this change. After the change, the `bloomFilterMightContain` will only be passed around in case of `BloomFilterApplicationStatus.FalsePositive` or `BloomFilterApplicationStatus.Success`.

Reply:
after the change, the `applyBloomFilter` function returning either `{ status: BloomFilterApplicationStatus.Skipped }`, or `{ status, bloomFilterMightContain };`, so basically, `applyResult.bloomFilterMightContain` is null if status is  `BloomFilterApplicationStatus.Skipped `.

So, remove the changes in ` applyBloomFilter`, let it return status only as before. 
And update the `createExistenceFilterMismatchInfoForTestingHooks` call to pass in the `bloomFilterMightContain` function based on `applyBloomFilter` result.
```
createExistenceFilterMismatchInfoForTestingHooks(
              applyResult.status,
              // applyResult.bloomFilterMightContain ?? null,     
              applyStatus !== BloomFilterApplicationStatus.Skipped ? applyResult.bloomFilterMightContain : null, 
              currentSize,
              watchChange.existenceFilter
            )
```

Reply:
But if `applyBloomFilter()` only returns the `status` then there is no way to pass along the `bloomFilterMightContain` to the testing hooks. Note that in this PR the testing hooks were augmented to include `bloomFilterMightContain` so that the integration tests can interrogate the bloom filter directly.

Reply:
awww, I see. We need the `bloomFilter`  object generated inside `applyBloomFilter` to get passed into the test helper.